var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -126,13 +160,42 @@ def configure_args(self):
             args.append("--with-kdreg=no")
 
         for fabric in self.fabrics:
```

Comment:
What about `for fabric in [f if isinstance(f, str) else f[0].value for f in self.fabrics]:`

And then you can jump straight to the `if "fabrics=" + fabric in self.spec:` check with a lot less fuss.

Reply:
@becker33 Thanks for the suggestion.  I have updated it.